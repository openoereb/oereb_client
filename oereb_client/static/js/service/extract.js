goog.provide('oereb.ExtractService');

goog.require('oereb');

/**
 * Angular service to request a specific extract.
 * @param {angular.$http} $http Angular service for HTTP requests.
 * @param {angular.$q} $q Angular service for deferrable objects.
 * @param {angular.Module.constant} oerebApplicationUrl Angular service for HTTP requests.
 * @constructor
 * @ngInject
 * @ngdoc service
 * @ngname ExtractService
 */
oereb.ExtractService = function($http, $q, oerebApplicationUrl) {

  this.$http_ = $http;
  this.$q_ = $q;
  this.applicationUrl_ = oerebApplicationUrl;

  this.extract_ = undefined;
  this.embeddable_ = undefined;

};

/**
 * Requests the extract for the specified EGRID.
 * @param {string} egrid The EGRID to request the extract for.
 * @returns {angular.$q.Promise} Promise for the EGRID request.
 */
oereb.ExtractService.prototype.queryExtractById = function(egrid) {
  this.extract_ = undefined;
  this.embeddable_ = undefined;
  var def = this.$q_.defer();
  this.$http_.get(this.applicationUrl_ + 'extract/reduced/json/geometry/' + egrid).then(
    function(response) {
      if (angular.isObject(response.data['GetExtractByIdResponse'])) {
        this.extract_ = response.data['GetExtractByIdResponse']['extract'];
        if (angular.isObject(response.data['GetExtractByIdResponse']['embeddable'])) {
          this.embeddable_ = response.data['GetExtractByIdResponse']['embeddable'];
        }
        def.resolve(response.data['GetExtractByIdResponse']);
      }
      else {
        def.reject('Invalid response format.');
      }
    }.bind(this),
    function(response) {
      var error = 'Requesting extract for ' + egrid + ' failed.';
      if (angular.isString(response.data)) {
        error += ' ' + response.data;
      }
      def.reject(error);
    }
  );
  return def.promise;
};

/**
 * Returns the extract if available.
 * @returns {Object|undefined} The extract object or undefined.
 */
oereb.ExtractService.prototype.getExtract = function() {
  return this.extract_;
};

/**
 * Returns the real estate data if available.
 * @returns {Object|undefined} The real estate data or undefined.
 */
oereb.ExtractService.prototype.getRealEstate = function() {
  if (angular.isDefined(this.getExtract())) {
    return this.getExtract()['RealEstate'];
  }
  return undefined;
};

/**
 * Returns list of concerned themes.
 * @returns {Array|undefined} The list of concerned themes.
 */
oereb.ExtractService.prototype.getConcernedThemes = function() {
  if (angular.isDefined(this.getExtract())) {
    return this.getExtract()['ConcernedTheme'];
  }
  return undefined;
};

/**
 * Returns list of not concerned themes.
 * @returns {Array|undefined} The list of not concerned themes.
 */
oereb.ExtractService.prototype.getNotConcernedThemes = function() {
  if (angular.isDefined(this.getExtract())) {
    return this.getExtract()['NotConcernedTheme'];
  }
  return undefined;
};

/**
 * Returns list of themes without data.
 * @returns {Array|undefined} The list of themes without data.
 */
oereb.ExtractService.prototype.getThemesWithoutData = function() {
  if (angular.isDefined(this.getExtract())) {
    return this.getExtract()['ThemeWithoutData'];
  }
  return undefined;
};

/**
 * Returns the public law restrictions of the specified topic.
 * @param {string} themeCode The code of the topic to return the restrictions for.
 * @returns {Array|undefined} The public law restrictions of the specified topic.
 */
oereb.ExtractService.prototype.getRestrictions = function(themeCode) {
  if (angular.isDefined(this.getRealEstate())) {
    var restrictions = [];
    angular.forEach(this.getRealEstate()['RestrictionOnLandownership'], function(restriction) {
      if (restriction['Theme']['Code'] === themeCode) {
        restrictions.push(restriction);
      }
    });
    return restrictions;
  }
  return undefined;
};

/**
 * Returns the legend entries of the specified topic.
 * @param {string} themeCode The code of the topic to return the restrictions for.
 * @returns {Object|undefined} The legend entries of the specified topic.
 */
oereb.ExtractService.prototype.getLegend = function(themeCode) {
  var restrictions = this.getRestrictions(themeCode);
  if (angular.isDefined(restrictions)) {
    var legendEntries = [];
    var legendGraphics = [];
    for (var i = 0; i < restrictions.length; i++) {
      var existing = false;
      for (var j = 0; j < legendEntries.length; j++) {
        if (legendEntries[j]['TypeCode'] === restrictions[i]['TypeCode']) {
          existing = true;
          legendEntries[j]['Area'] += restrictions[i]['Area'];
          legendEntries[j]['PartInPercent'] += restrictions[i]['PartInPercent'];
          break;

        }

      }
      if (!existing) {
        legendEntries.push({
          'TypeCode': restrictions[i]['TypeCode'],
          'Information': restrictions[i]['Information'],
          'Area': restrictions[i]['Area'],
          'PartInPercent': restrictions[i]['PartInPercent'],
          'SymbolRef': restrictions[i]['SymbolRef']
        });
      }
      this.addIfNotContains_(restrictions[i]['Map']['LegendAtWeb'], legendGraphics);
    }
    return {
      'entries': legendEntries,
      'graphics': legendGraphics
    };
  }
  return undefined;
};

/**
 * Adds the item to the specified target array if it's not already contained. `angular.equals` is used for the
 * comparison.
 * @param {Object|Array|string|number} item The element to be added.
 * @param {Array} target The target array to add the element to.
 * @private
 */
oereb.ExtractService.prototype.addIfNotContains_ = function(item, target) {
  if (angular.isArray(target)) {
    for (var i = 0; i < target.length; i++) {
      if (angular.equals(target[i], item)) {
        return;
      }
    }
    target.push(item);
  }
};

/**
 * Adds the document to the specified target array if it's not already contained. `angular.equals` is used for
 * the comparison.
 * @param {Object} document The document to be added.
 * @param {Array} target The target array to add the element to.
 * @private
 */
oereb.ExtractService.prototype.addDocumentIfNotContained_ = function(document, target) {
  for (var i = 0; i < target.length; i++) {
    // Check if document already exists (only use 'Title', 'OfficialNumber' and 'TextAtWeb' for comparison)
    if (
      angular.equals(target[i]['Title'], document['Title']) &&
      angular.equals(target[i]['OfficialNumber'], document['OfficialNumber']) &&
      angular.equals(target[i]['TextAtWeb'], document['TextAtWeb'])
    ) {
      // Add missing article numbers if document already exists
      for (var j = 0; j < document['ArticleNumber'].length; j++) {
        this.addIfNotContains_(document['ArticleNumber'][j], target[i]['ArticleNumber']);
      }
      // Add missing articles if document already exists
      for (var k = 0; k < document['Article'].length; k++) {
        this.addIfNotContains_(document['Article'][k], target[i]['Article']);
      }
      return;
    }
  }
  // Add document if it is missing
  target.push(document);
};

/**
 * Add each listed document to the target array if it's not already contained.
 * @param {Array} documents The documents to be added.
 * @param {Array} target The target array.
 * @private
 */
oereb.ExtractService.prototype.addDocumentsIfNotContained_ = function(documents, target) {
  // Iterate documents to be added
  for (var i = 0; i < documents.length; i++) {
    // Create document object with necessary properties and check if it needs to be added
    this.addDocumentIfNotContained_({
      'Title': documents[i]['Title'],
      'OfficialNumber': documents[i]['OfficialNumber'],
      'ArticleNumber': documents[i]['ArticleNumber'] || [],
      'Article': documents[i]['Article'] || [],
      'TextAtWeb': documents[i]['TextAtWeb']
    }, target);
    // Do the same for possible references
    var references = documents[i]['Reference'];
    if (angular.isArray(references)) {
      this.addDocumentsIfNotContained_(references, target);
    }
  }
};

/**
 * Returns the legal documents of the specified topic.
 * @param {string} themeCode The code of the topic to return the restrictions for.
 * @returns {Object|undefined} The legal documents of the specified topic.
 */
oereb.ExtractService.prototype.getDocuments = function(themeCode) {
  // Get the restrictions for the specified topic
  var restrictions = this.getRestrictions(themeCode);
  if (angular.isArray(restrictions)) {
    var legalProvisions = [];
    var documents = [];
    // Iterate the resulting restrictions
    for (var i = 0; i < restrictions.length; i++) {
      // Iterate the legal provisions for each restriction
      var legalProvision = restrictions[i]['LegalProvisions'];
      for (var j = 0; j < legalProvision.length; j++) {
        // Create legal provision object with necessary properties and check if it needs to be added
        this.addDocumentIfNotContained_({
          'Title': legalProvision[j]['Title'],
          'OfficialNumber': legalProvision[j]['OfficialNumber'],
          'ArticleNumber': legalProvision[j]['ArticleNumber'] || [],
          'Article': legalProvision[j]['Article'] || [],
          'TextAtWeb': legalProvision[j]['TextAtWeb']
        }, legalProvisions);
        // Add possible references to the documents
        var references = legalProvision[j]['Reference'];
        if (angular.isArray(references)) {
          this.addDocumentsIfNotContained_(references, documents);
        }
      }
    }
    // Add possible references for the real estate to the documents
    this.addDocumentsIfNotContained_(this.getRealEstate()['Reference'], documents);
    return {
      'LegalProvisions': legalProvisions,
      'Documents': documents
    };
  }
  return undefined;
};

/**
 * Returns the responsible offices of the specified topic.
 * @param {string} themeCode The code of the topic to return the responsible offices for.
 * @returns {Array|undefined} The responsible offices of the specified topic.
 */
oereb.ExtractService.prototype.getResponsibleOffices = function(themeCode) {
  var restrictions = this.getRestrictions(themeCode);
  if (angular.isArray(restrictions)) {
    var offices = [];
    for (var i = 0; i < restrictions.length; i++) {
      this.addIfNotContains_(restrictions[i]['ResponsibleOffice'], offices);
    }
    return offices;
  }
  return undefined;
};

/**
 * Creates a view service definition using the restriction's theme code and view service URL.
 * @param {string} themeCode The restriction's theme code.
 * @param {string} viewServiceUrl The restriction's view service URL.
 * @returns {Object} The created view service definition.
 * @private
 */
oereb.ExtractService.prototype.getViewServiceFromUrl_ = function(themeCode, viewServiceUrl) {
  var parts = viewServiceUrl.split('?');
  var url = parts[0];
  var definition = {
    'topic': themeCode,
    'url': url,
    'params': {}
  };
  var params = parts[1].split('&');
  for (var i = 0; i < params.length; i++) {
    var param = params[i].split('=');
    if (param[0].toUpperCase() === 'LAYERS') {
      definition['params']['LAYERS'] = param[1];
    }
    else if (param[0].toUpperCase() === 'STYLES') {
      definition['params']['STYLES'] = param[1];
    }
    else if (param[0].toUpperCase() === 'VERSION') {
      definition['params']['VERSION'] = param[1];
    }
  }
  return definition;
};

/**
 * Returns a unique list of view services available in the extract.
 * @returns {Array} A unique list of view service objects.
 */
oereb.ExtractService.prototype.getViewServices = function() {
  var viewServices = [];
  var realEstate = this.getRealEstate();
  if (angular.isDefined(realEstate)) {
    var restrictions = realEstate['RestrictionOnLandownership'];
    if (angular.isArray(restrictions)) {
      for (var i = 0; i < restrictions.length; i++) {
        var url = restrictions[i]['Map']['ReferenceWMS'];
        if (angular.isDefined(url)) {
          this.addIfNotContains_(
            this.getViewServiceFromUrl_(restrictions[i]['Theme']['Code'], url),
            viewServices
          );
        }
      }
    }
  }
  return viewServices;
};

/**
 * Returns the exclusions of liability for the current extract.
 * @returns {Array} The exclusions of liability.
 */
oereb.ExtractService.prototype.getExclusionsOfLiability = function() {
  if (angular.isDefined(this.getExtract()) && angular.isArray(this.getExtract()['ExclusionOfLiability'])) {
    return this.getExtract()['ExclusionOfLiability'];
  }
  return [];
};

/**
 * Returns the glossary for the current extract.
 * @returns {Array} The exclusions of liability.
 */
oereb.ExtractService.prototype.getGlossary = function() {
  if (angular.isDefined(this.getExtract()) && angular.isArray(this.getExtract()['Glossary'])) {
    return this.getExtract()['Glossary'];
  }
  return [];
};

/**
 * Returns the embeddable if available.
 * @returns {Object|undefined} The extract object or undefined.
 */
oereb.ExtractService.prototype.getEmbeddable = function() {
  return this.embeddable_;
};

oereb.module.service('ExtractService', oereb.ExtractService);
