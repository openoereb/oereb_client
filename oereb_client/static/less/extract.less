/**
 * Extract styling
 */

@import "variables";

div#extract {
  width: @menu-with;
  max-width: 100%;
  position: fixed;
  top: 0;
  left: 0;
  height: 0;
  padding-top: @header-height;
  background-color: #fff;
  transition: height @animation-duration;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: stretch;
  div.extract-wrapper {
    z-index: 1;
    background-color: @menu-background-color;
    overflow-y: scroll;
    flex: 1;
    padding-bottom: @default-padding;
    margin-top: @default-padding;
  }
  span.extract-close {
    z-index: 2;
    display: block;
    float: right;
    color: @brand-primary;
    opacity: 1;
  }
  span.extract-close:hover {
    color: #000000;
    opacity: 1;
  }
  .panel-default > .panel-heading .badge {
    background-color: @brand-primary;
  }
  .collapse-extract {
    padding-top: @default-padding;
    .btn {
      border-bottom-width: 0;
    }
  }
}

div#extract.loading {
  height: @loading-indicator-height;
}

div#extract.active {
  height: 100%;
}

div#extract.information {
  box-shadow: @menu-shadow;
}