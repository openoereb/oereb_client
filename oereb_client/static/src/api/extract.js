export function queryExtractById(applicationUrl, egrid) {
    const url = new URL(applicationUrl + 'extract/reduced/json/geometry/' + egrid);
    url.searchParams.append('_dc', new Date().getTime());
    return fetch(url).then(response => response.json());
}

export function sanitizeTopicCode(code) {
    return code.replace('.', '').replace('-', '').replace('_', '');
}

export function groupRestrictionsByTopic(retstrictions) {
    const restrictionsByTopic = {};
    retstrictions.forEach((restriction) => {
        const code = sanitizeTopicCode(restriction.Theme.Code);
        if (!restrictionsByTopic.hasOwnProperty(code)) {
            restrictionsByTopic[code] = [];
        }
        restrictionsByTopic[code].push(restriction);
    });
    return restrictionsByTopic;
}
