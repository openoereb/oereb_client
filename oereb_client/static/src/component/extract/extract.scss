@import "../../oereb_client.scss";

div.oereb-client-extract-wrapper {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    background-color: $menu-background-color;
    box-shadow: $default-shadow;
    overflow: hidden;
    transition: height $animation-duration linear;

    div.oereb-client-extract {
        width: 100%;
        height: 100%;
        padding-top: $header-height;
        overflow: hidden;
    }
    
    div.oereb-client-extract.data {
        .category:not(.show) {
            border: 0px;
        }
    
        & > div.accordion {
            overflow-y: scroll;
    
            button.accordion-button:focus,
            button.accordion-button:not(.collapsed) {
                box-shadow: none;
            }
        }

        @include media-breakpoint-up(xl) {
            button.collapse-extract {
                display: none !important;
            }
        }
    }
}

div.oereb-client-extract-wrapper.hidden {
    height: $header-height;
}

div.oereb-client-extract-wrapper.loading {
    height: $header-height + 100px;
}

div.oereb-client-extract-wrapper.shown {
    height: 100%;
}

div.oereb-client-extract-wrapper.collapsed {
    height: $header-height + 50px;
}

@include media-breakpoint-up(xl) {
    div.oereb-client-extract-wrapper {
        width: $menu-with;
    }
}