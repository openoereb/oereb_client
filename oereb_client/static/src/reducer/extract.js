import { createSlice } from "@reduxjs/toolkit";

export const extractSlice = createSlice({
    name: 'extract',
    initialState: {
        loading: false,
        visible: false,
        error: false,
        egrid: null,
        data: {}
    },
    reducers: {
        loadExtract: (state, action) => {
            console.log('Extract loading');
            if (!state.loading) {
                state.loading = true;
                state.visible = false;
                state.error = false;
                state.egrid = action.payload.egrid;
                state.data = {};
            }
        },
        showExtract: (state, action) => {
            state.loading = false;
            state.visible = true;
            state.error = false;
            state.data = action.payload.extract;
            console.log('Extract loaded', state.data);
        },
        showError: (state, action) => {
            state.loading = false;
            state.visible = false;
            state.error = true;
            state.data = {};
            console.log('Extract failed');
        },
        hideExtract: (state, action) => {
            state.loading = false;
            state.visible = false;
            state.error = false;
            state.egrid = null;
            state.data = {};
            console.log('Extract closed');
        }
    }
});

export const { loadExtract, showExtract, showError, hideExtract } = extractSlice.actions;

export default extractSlice.reducer;
