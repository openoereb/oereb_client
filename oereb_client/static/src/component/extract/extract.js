import './extract.scss';

import React from 'react';
import { useSelector } from 'react-redux';
import OerebExtractLoading from './extract_loading';
import OerebExtractError from './extract_error';
import OerebExtractData from './extract_data';

function OerebExtract(props) {
    const extract = useSelector((state) => state.extract);

    function getContent(cfg) {
        if (cfg.loading) {
            return (
                <OerebExtractLoading />
            );
        }
        else if (cfg.error) {
            return (
                <OerebExtractError />
            );
        }
        else if (cfg.visible) {
            return (
                <OerebExtractData data={extract.data.GetExtractByIdResponse.extract} />
            );
        }
        else {
            return null;
        }
    }

    const content = getContent(extract);

    return (
        <div class="oereb-client-extract">
            {content}
        </div>
    );
}

export default OerebExtract;
