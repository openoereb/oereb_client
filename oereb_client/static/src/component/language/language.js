import PropTypes from 'prop-types';
import React from "react";
import {withNamespaces} from 'react-i18next';
import {useDispatch, useSelector} from "react-redux";

import {setLanguage} from "../../reducer/language";

const OerebLanguage = function({t}) {
  const dispatch = useDispatch();
  const language = useSelector((state) => state.language);
  const currentLanguage = language.current;
  const availableLanguages = language.available;

  const selectLanguage = function(lang) {
    dispatch(setLanguage(lang));
  };

  const languages = availableLanguages.map((language, key) =>
    <li key={key}>
      <button className="dropdown-item" onClick={selectLanguage.bind(this, language)}>
        {language.toUpperCase()}
      </button>
    </li>
  );

  return (
    <div>
      <button className="btn btn-outline-secondary dropdown-toggle"
        type="button"
        data-bs-toggle="dropdown"
        aria-expanded="false"
        title={t('menu.language.title')}>
        {currentLanguage.toUpperCase()}
      </button>
      <ul className="dropdown-menu">
        {languages}
      </ul>
    </div>
  );
};

OerebLanguage.propTypes = {
  t: PropTypes.func.isRequired
};

export default withNamespaces()(OerebLanguage);
