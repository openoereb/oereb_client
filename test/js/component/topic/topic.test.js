import {mount} from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";
import {act} from "react-dom/test-utils";
import {Provider} from "react-redux";

import OerebTopic from "../../../../oereb_client/static/src/component/topic/topic";
import {initLanguages} from "../../../../oereb_client/static/src/reducer/language";
import {groupRestrictionsByTopic} from "../../../../oereb_client/static/src/request/extract";
import MainStore from "../../../../oereb_client/static/src/store/main";
import extract from "../../../../samples/extract.json";

describe('topic component', () => {

  let component;

  beforeEach(() => {
    act(() => {
      MainStore.dispatch(initLanguages({
        default: 'de',
        available: ['de']
      }));
    });
    const restrictions = groupRestrictionsByTopic(
      extract.GetExtractByIdResponse.extract.RealEstate.RestrictionOnLandownership,
      extract.GetExtractByIdResponse.extract.ConcernedTheme
    )['chStatischeWaldgrenzen']['inForce'];
    component = mount(
      <Provider store={MainStore}>
        <OerebTopic restrictions={restrictions} />
      </Provider>
    );
  });

  it('should render topic', () => {
    expect(toJson(component)).toMatchSnapshot();
    component.find('button').simulate('click');
    expect(toJson(component)).toMatchSnapshot();
  });

});