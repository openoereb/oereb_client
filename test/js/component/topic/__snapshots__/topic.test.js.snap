// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`topic component should render topic 1`] = `
<DocumentFragment>
  <div
    class="accordion-item oereb-client-topic"
  >
    <h2
      class="accordion-header"
    >
      <button
        class="accordion-button pt-2 pb-2 collapsed"
        type="button"
      >
        <div
          class="oereb-topic-lawstatus"
        >
          <small
            class="fst-italic"
          >
            Rechtskräftig
          </small>
        </div>
      </button>
    </h2>
    <div
      class="accordion-collapse collapse"
    >
      <div
        class="accordion-body"
      >
        <div
          class="row align-items-center mb-2"
        >
          <div
            class="col-3"
          >
            <small>
              extract.topic.opacity:
            </small>
          </div>
          <div
            class="col-2 pe-2 text-end"
          >
            <small>
              100%
            </small>
          </div>
          <div
            class="col-7"
          >
            <input
              class="form-range align-middle"
              max="100"
              min="0"
              type="range"
              value="100"
            />
          </div>
        </div>
        <table
          class="table table-sm mb-2 oereb-client-legend"
        >
          <thead>
            <tr>
              <th>
                extract.topic.legend.type
              </th>
              <th />
              <th
                class="text-end text-nowrap"
              >
                extract.topic.legend.part
              </th>
              <th
                class="text-end text-nowrap"
              >
                extract.topic.legend.part_in_percent
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                class="oereb-client-symbol-cell"
              >
                <img
                  class="oereb-client-symbol"
                  src="samples/static/symbol2.png"
                />
              </td>
              <td
                class="oereb-client-legend-text"
              >
                Belastet, untersuchungsbedürftig
              </td>
              <td
                class="oereb-client-legend-text text-end text-nowrap"
              >
                202 m
              </td>
              <td
                class="oereb-client-legend-text text-end text-nowrap"
              />
            </tr>
          </tbody>
        </table>
        <dl
          class="oereb-client-documents mb-2"
        >
          <dt>
            extract.topic.document.legal_provisions
          </dt>
          <dd
            class="oereb-client-document ms-2"
          >
            <span>
              Baugesetz, 3891.100
            </span>
            <ul
              class="ps-2 mb-2"
            >
              <li>
                <a
                  href="https://oereb-gr-preview.000.ch/api/attachments/197"
                  rel="noreferrer"
                  target="_blank"
                >
                  https://oereb-gr-preview.000.ch/api/attachments/197
                </a>
              </li>
              <li>
                <a
                  href="https://oereb-gr-preview.000.ch/api/attachments/198"
                  rel="noreferrer"
                  target="_blank"
                >
                  https://oereb-gr-preview.000.ch/api/attachments/198
                </a>
              </li>
            </ul>
          </dd>
          <dd
            class="oereb-client-document ms-2"
          >
            <span>
              Revision Ortsplanung, 07.447
            </span>
            <ul
              class="ps-2 mb-2"
            >
              <li>
                <a
                  href="https://oereb-gr-preview.000.ch/api/attachments/213"
                  rel="noreferrer"
                  target="_blank"
                >
                  https://oereb-gr-preview.000.ch/api/attachments/213
                </a>
              </li>
              <li>
                <a
                  href="https://oereb-gr-preview.000.ch/api/attachments/214"
                  rel="noreferrer"
                  target="_blank"
                >
                  https://oereb-gr-preview.000.ch/api/attachments/214
                </a>
              </li>
            </ul>
          </dd>
          <dt>
            extract.topic.document.laws
          </dt>
          <dd
            class="oereb-client-document ms-2"
          >
            <span>
              Raumplanungsverordnung für den Kanton Graubünden (KRVO), BR 801.110
            </span>
            <ul
              class="ps-2 mb-2"
            >
              <li>
                <a
                  href="https://www.gr-lex.gr.ch/frontend/versions/pdf_file_with_annex/2934?locale=de"
                  rel="noreferrer"
                  target="_blank"
                >
                  https://www.gr-lex.gr.ch/frontend/versions/pdf_file_with_annex/2934?locale=de
                </a>
              </li>
            </ul>
          </dd>
          <dd
            class="oereb-client-document ms-2"
          >
            <span>
              Bundesgesetz über die Raumplanung (Raumplanungsgesetz, RPG) (RPG), SR 700
            </span>
            <ul
              class="ps-2 mb-2"
            >
              <li>
                <a
                  href="http://www.lexfind.ch/dtah/167348/2"
                  rel="noreferrer"
                  target="_blank"
                >
                  http://www.lexfind.ch/dtah/167348/2
                </a>
              </li>
            </ul>
          </dd>
          <dd
            class="oereb-client-document ms-2"
          >
            <span>
              Raumplanungsgesetz für den Kanton Graubünden (KRG), BR 801.100
            </span>
            <ul
              class="ps-2 mb-2"
            >
              <li>
                <a
                  href="https://www.gr-lex.gr.ch/frontend/versions/pdf_file_with_annex/2936?locale=de"
                  rel="noreferrer"
                  target="_blank"
                >
                  https://www.gr-lex.gr.ch/frontend/versions/pdf_file_with_annex/2936?locale=de
                </a>
              </li>
            </ul>
          </dd>
          <dt>
            extract.topic.document.hints
          </dt>
          <dd
            class="oereb-client-document ms-2"
          >
            <span>
              Guter Ratschlag
            </span>
            <ul
              class="ps-2 mb-2"
            >
              <li>
                <a
                  href="http://example.com/goodhint1"
                  rel="noreferrer"
                  target="_blank"
                >
                  http://example.com/goodhint1
                </a>
              </li>
              <li>
                <a
                  href="http://example.com/goodhint2"
                  rel="noreferrer"
                  target="_blank"
                >
                  http://example.com/goodhint2
                </a>
              </li>
            </ul>
          </dd>
          <dd
            class="oereb-client-document ms-2"
          >
            <span>
              Seht guter Ratschlag mit langem titel: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras convallis venenatis posuere. Nunc sollicitudin fermentum bibendum. Nulla ut massa enim. Sed rhoncus euismod venenatis
            </span>
            <ul
              class="ps-2 mb-2"
            >
              <li>
                <a
                  href="http://example.com/goodhint3"
                  rel="noreferrer"
                  target="_blank"
                >
                  http://example.com/goodhint3
                </a>
              </li>
            </ul>
          </dd>
        </dl>
        <dl
          class="oereb-client-responsible-office border-top pt-2 mb-2"
        >
          <dt>
            extract.topic.responsible_office
          </dt>
          <dd
            class="ms-2"
          >
            <a
              href="https://www.bav.admin.ch/bav/de/home.html"
              rel="noreferrer"
              target="_blank"
            >
              Bundesamt für Verkehr BAV
            </a>
          </dd>
        </dl>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`topic component should render topic 2`] = `
<DocumentFragment>
  <div
    class="accordion-item oereb-client-topic"
  >
    <h2
      class="accordion-header"
    >
      <button
        class="accordion-button pt-2 pb-2"
        type="button"
      >
        <div
          class="oereb-topic-lawstatus"
        >
          <small
            class="fst-italic"
          >
            Rechtskräftig
          </small>
        </div>
      </button>
    </h2>
    <div
      class="accordion-collapse collapsing"
      style="height: 0px;"
    >
      <div
        class="accordion-body"
      >
        <div
          class="row align-items-center mb-2"
        >
          <div
            class="col-3"
          >
            <small>
              extract.topic.opacity:
            </small>
          </div>
          <div
            class="col-2 pe-2 text-end"
          >
            <small>
              100%
            </small>
          </div>
          <div
            class="col-7"
          >
            <input
              class="form-range align-middle"
              max="100"
              min="0"
              type="range"
              value="100"
            />
          </div>
        </div>
        <table
          class="table table-sm mb-2 oereb-client-legend"
        >
          <thead>
            <tr>
              <th>
                extract.topic.legend.type
              </th>
              <th />
              <th
                class="text-end text-nowrap"
              >
                extract.topic.legend.part
              </th>
              <th
                class="text-end text-nowrap"
              >
                extract.topic.legend.part_in_percent
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                class="oereb-client-symbol-cell"
              >
                <img
                  class="oereb-client-symbol"
                  src="samples/static/symbol2.png"
                />
              </td>
              <td
                class="oereb-client-legend-text"
              >
                Belastet, untersuchungsbedürftig
              </td>
              <td
                class="oereb-client-legend-text text-end text-nowrap"
              >
                202 m
              </td>
              <td
                class="oereb-client-legend-text text-end text-nowrap"
              />
            </tr>
          </tbody>
        </table>
        <dl
          class="oereb-client-documents mb-2"
        >
          <dt>
            extract.topic.document.legal_provisions
          </dt>
          <dd
            class="oereb-client-document ms-2"
          >
            <span>
              Baugesetz, 3891.100
            </span>
            <ul
              class="ps-2 mb-2"
            >
              <li>
                <a
                  href="https://oereb-gr-preview.000.ch/api/attachments/197"
                  rel="noreferrer"
                  target="_blank"
                >
                  https://oereb-gr-preview.000.ch/api/attachments/197
                </a>
              </li>
              <li>
                <a
                  href="https://oereb-gr-preview.000.ch/api/attachments/198"
                  rel="noreferrer"
                  target="_blank"
                >
                  https://oereb-gr-preview.000.ch/api/attachments/198
                </a>
              </li>
            </ul>
          </dd>
          <dd
            class="oereb-client-document ms-2"
          >
            <span>
              Revision Ortsplanung, 07.447
            </span>
            <ul
              class="ps-2 mb-2"
            >
              <li>
                <a
                  href="https://oereb-gr-preview.000.ch/api/attachments/213"
                  rel="noreferrer"
                  target="_blank"
                >
                  https://oereb-gr-preview.000.ch/api/attachments/213
                </a>
              </li>
              <li>
                <a
                  href="https://oereb-gr-preview.000.ch/api/attachments/214"
                  rel="noreferrer"
                  target="_blank"
                >
                  https://oereb-gr-preview.000.ch/api/attachments/214
                </a>
              </li>
            </ul>
          </dd>
          <dt>
            extract.topic.document.laws
          </dt>
          <dd
            class="oereb-client-document ms-2"
          >
            <span>
              Raumplanungsverordnung für den Kanton Graubünden (KRVO), BR 801.110
            </span>
            <ul
              class="ps-2 mb-2"
            >
              <li>
                <a
                  href="https://www.gr-lex.gr.ch/frontend/versions/pdf_file_with_annex/2934?locale=de"
                  rel="noreferrer"
                  target="_blank"
                >
                  https://www.gr-lex.gr.ch/frontend/versions/pdf_file_with_annex/2934?locale=de
                </a>
              </li>
            </ul>
          </dd>
          <dd
            class="oereb-client-document ms-2"
          >
            <span>
              Bundesgesetz über die Raumplanung (Raumplanungsgesetz, RPG) (RPG), SR 700
            </span>
            <ul
              class="ps-2 mb-2"
            >
              <li>
                <a
                  href="http://www.lexfind.ch/dtah/167348/2"
                  rel="noreferrer"
                  target="_blank"
                >
                  http://www.lexfind.ch/dtah/167348/2
                </a>
              </li>
            </ul>
          </dd>
          <dd
            class="oereb-client-document ms-2"
          >
            <span>
              Raumplanungsgesetz für den Kanton Graubünden (KRG), BR 801.100
            </span>
            <ul
              class="ps-2 mb-2"
            >
              <li>
                <a
                  href="https://www.gr-lex.gr.ch/frontend/versions/pdf_file_with_annex/2936?locale=de"
                  rel="noreferrer"
                  target="_blank"
                >
                  https://www.gr-lex.gr.ch/frontend/versions/pdf_file_with_annex/2936?locale=de
                </a>
              </li>
            </ul>
          </dd>
          <dt>
            extract.topic.document.hints
          </dt>
          <dd
            class="oereb-client-document ms-2"
          >
            <span>
              Guter Ratschlag
            </span>
            <ul
              class="ps-2 mb-2"
            >
              <li>
                <a
                  href="http://example.com/goodhint1"
                  rel="noreferrer"
                  target="_blank"
                >
                  http://example.com/goodhint1
                </a>
              </li>
              <li>
                <a
                  href="http://example.com/goodhint2"
                  rel="noreferrer"
                  target="_blank"
                >
                  http://example.com/goodhint2
                </a>
              </li>
            </ul>
          </dd>
          <dd
            class="oereb-client-document ms-2"
          >
            <span>
              Seht guter Ratschlag mit langem titel: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras convallis venenatis posuere. Nunc sollicitudin fermentum bibendum. Nulla ut massa enim. Sed rhoncus euismod venenatis
            </span>
            <ul
              class="ps-2 mb-2"
            >
              <li>
                <a
                  href="http://example.com/goodhint3"
                  rel="noreferrer"
                  target="_blank"
                >
                  http://example.com/goodhint3
                </a>
              </li>
            </ul>
          </dd>
        </dl>
        <dl
          class="oereb-client-responsible-office border-top pt-2 mb-2"
        >
          <dt>
            extract.topic.responsible_office
          </dt>
          <dd
            class="ms-2"
          >
            <a
              href="https://www.bav.admin.ch/bav/de/home.html"
              rel="noreferrer"
              target="_blank"
            >
              Bundesamt für Verkehr BAV
            </a>
          </dd>
        </dl>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
